File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
  public int firstMissingPositive(int[] nums) {
    HashSet positives = new HashSet<Integer>();
    for (int n : nums) {
      if (n > 0) positives.add(n);
    }
    int missing = 1;
    while (positives.contains(missing)) missing++;
    return missing;
  }
}

File: 0041-first-missing-positive/0041-first-missing-positive.go
func firstMissingPositive(nums []int) int {
  positives := map[int]bool{}
  for _, n := range nums {
    if n > 0 {
      positives[n] = true
    }
  }
  missing := 1
  for true {
    if _, ok := positives[missing]; ok {
      missing++
    } else {
      break
    }
  }
  return missing
}